tests/Jellyfin.Naming.Tests/AudioBook/AudioBookListResolverTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Linq;
+using Emby.Naming.AudioBook;
+using Emby.Naming.Common;
+using MediaBrowser.Model.IO;
+using Xunit;
+
+namespace Jellyfin.Naming.Tests.AudioBook
+{
+    public class AudioBookListResolverTests
+    {
+        private readonly NamingOptions _namingOptions = new NamingOptions();
+
+        [Fact]
+        public void TestStackAndExtras()
+        {
+            // No stacking here because there is no part/disc/etc
+            var files = new[]
+            {
+                "Harry Potter and the Deathly Hallows/Part 1.mp3",
+                "Harry Potter and the Deathly Hallows/Part 2.mp3",
+                "Harry Potter and the Deathly Hallows/book.nfo",
+
+                "Batman/Chapter 1.mp3",
+                "Batman/Chapter 2.mp3",
+                "Batman/Chapter 3.mp3",
+            };
+
+            var resolver = GetResolver();
+
+            var result = resolver.Resolve(files.Select(i => new FileSystemMetadata
+            {
+                IsDirectory = false,
+                FullName = i
+            }).ToList()).ToList();
+
+            Assert.Equal(2, result[0].Files.ToList().Count);
```

Comment:
```suggestion
            Assert.Equal(2, result[0].Files.Count);
```

Reply:
xunit shouted at me when I tried that, let me test that again